src/core/dev/allocator.cpp

```diff
@@ -255,36 +255,91 @@ xlio_registrator::~xlio_registrator()
 }
 
 bool xlio_registrator::register_memory(void *data, size_t size)
+{
+    bool rc = true;
+#ifdef DEFINED_DPCP_PATH_RX_OR_TX
+    rc &= register_memory_dpcp(data, size);
+#endif // DEFINED_DPCP_PATH_RX_OR_TX
+#ifndef DEFINED_DPCP_PATH_RX_AND_TX
+    rc &= register_memory_doca(data, size);
+#endif // !DEFINED_DPCP_PATH_RX_AND_TX
```

Comment:
It is my feeling but following constructions more readable
and allows to easy search by exact pattern
```
#if defined(DEFINED_DPCP_PATH_RX) || defined(DEFINED_DPCP_PATH_TX)
```
or
```
#if defined(DEFINED_DPCP_PATH_RX) && defined(DEFINED_DPCP_PATH_TX)
```

Reply:
I agree as well. We need to see how we can work out classe such as ring_simple that from one point has many common elements on the other many non common. Maybe some common base?

Reply:
> I agree as well. We need to see how we can work out classe such as ring_simple that from one point has many common elements on the other many non common. Maybe some common base?

ring->ring_slave->[ring_simple|ring_tap]
      ->ring_bond->[ring_eth|ring_ib]

Reply:
It requires redesign and requires effort approval from management.

Reply:
As discussed with Tomer, having clear ifdefs like DPCP_PATH_TX | RX | ANY | NONE is more readable then one macros with numbers.